frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx

```diff
@@ -0,0 +1,491 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { Helmet } from 'react-helmet';
+import { ActionGroup, Button } from '@patternfly/react-core';
+import { safeLoad } from 'js-yaml';
+import * as classNames from 'classnames';
+
+import {
+  ButtonBar,
+  Dropdown,
+  Firehose,
+  history,
+  LoadError,
+  LoadingBox,
+  StatusBox,
+} from '../../utils';
+import {
+  getAlertManagerConfig,
+  patchAlertManagerConfig,
+  receiverTypes,
+} from '../alert-manager-utils';
+import { K8sResourceKind } from '../../../module/k8s';
+import {
+  AlertManagerConfig,
+  AlertManagerReceiver,
+  AlertManagerRoute,
+} from '../alert-manager-config';
+import { RoutingLabelEditor } from './routing-labels-editor';
+import * as PagerDutyForm from './pagerduty-receiver-form';
+import * as WebhookForm from './webhook-receiver-form';
+import * as EmailForm from './email-receiver-form';
+import * as SlackForm from './slack-receiver-form';
+import { coFetchJSON } from '../../../co-fetch';
+
+export const equalOrEmpty = (formValue, globalValue) =>
+  formValue === globalValue || _.isEmpty(formValue);
+export const notEqualAndNotEmpty = (formValue, globalValue) =>
+  formValue !== globalValue && !_.isEmpty(formValue);
```

Comment:
I realize these eliminate some duplicate code, but I think they're too low level. I'd say either replace them with something more semantic, or just write out each condition in full where it's needed.

Do we need to call `_.isEmpty` here or should it be `formValue === undefined`?

Reply:
Could you pls provide example of 'more semantic'? I use these functions in three receiver forms, I believe it is good coding practice to reduce duplicate code (DRY principle), plus it is very clear what these functions do -makes the code easier to follow IMO.

Reply:
> Do we need to call _.isEmpty here or should it be formValue === undefined?

`_.isEmpty` because formValue may have an initial value of ''.